plonky3/src/prover/mod.rs

```diff
@@ -92,18 +113,52 @@ mod tests {
         let witness_callback = pipeline.witgen_callback().unwrap();
         let witness = pipeline.compute_witness().unwrap();
 
-        let proof = Plonky3Prover::new(&pil).prove(&witness, witness_callback);
+        let proof = Plonky3Prover::new(&pil).prove(&witness, witness_callback, None);
+
+        assert!(proof.is_ok());
+    }
+
+    fn run_test_goldilocks_publics(pil:& str,  publics: Vec<Goldilocks>) {
+        let mut pipeline = Pipeline::<GoldilocksField>::default().from_pil_string(pil.to_string());
+
+        let pil = pipeline.compute_optimized_pil().unwrap();
+        let witness_callback = pipeline.witgen_callback().unwrap();
+        let witness = pipeline.compute_witness().unwrap();
+
+        let proof = Plonky3Prover::new(&pil).prove(&witness, witness_callback, Some(publics));
```

Comment:
Oh I see, so you added optional inputs so that you can test with malicious inputs? In that case I think we should keep the prover as it was, and instead call the verifier with the malicious inputs.